velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -436,6 +437,70 @@ TEST_F(ArithmeticTest, isNanDouble) {
   EXPECT_EQ(false, isNan(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, normalizeNanFloat) {
+  const auto normalizeNan = [&](std::optional<float> a) {
+    return evaluateOnce<float>("normalize_nan(c0)", a);
+  };
+
+  const auto hasher = [&](std::optional<float> f) {
```

Comment:
This shouldn't be needed. As long as you verified that binary representation is the same, the hash is guaranteed to be the same. If not, it is a bug in a hash function, but we are not testing hash function here.